src/helpers.rs

```diff
@@ -459,3 +460,16 @@ fn bytes_to_os_str<'tcx, 'a>(bytes: &'a[u8]) -> InterpResult<'tcx, &'a OsStr> {
         .map_err(|_| err_unsup_format!("{:?} is not a valid utf-8 string", bytes))?;
     Ok(&OsStr::new(s))
 }
+
+pub fn try_into_host_usize(i: impl Into<u128>) -> Option<usize> {
+    let i: u128 = i.into();
+    if i > usize::max_value() as u128 {
+        None
+    } else {
+        Some(i as usize)
+    }
+}
+
+pub fn try_from_host_usize<T: TryFrom<u128>>(i: usize) -> Option<T> {
+    T::try_from(i as u128).ok()
+}
```

Comment:
I am so sorry, it turns out we don't need these! All the `TryFrom` instances seem to actually exist for `usize` and `isize` (it's just `From` instances that are missing).

So you should be able to use `usize::try_from(...)` to convert anything into a host usize.
And we should probably rename the `to_usize`/`to_isize` methods on `Scalar` to `target_{u,i}size`, this is somewhat confusing...